sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,337 @@
+        vals_addr_invoicing = so_vals["address_invoicing"]
+        vals_addr_shipping = so_vals["address_shipping"]
+        for address_field in (
+            (vals_addr_shipping, "partner_shipping_id", "address_shipping", "delivery"),
+            (vals_addr_invoicing, "partner_invoice_id", "address_invoicing", "invoice"),
+        ):
+            addr = address_field[0]
+            field = address_field[1]
+            vals_key = address_field[2]
+            type_di = address_field[3]
+            if addr.get("state_code"):
+                addr["state_id"] = self.env["res.country.state"].search(
+                    [("code", "=", addr.get("state_code"))]
+                )
+                del addr["state_code"]
+            addr["country_id"] = self.env["res.country"].search(
+                [("code", "=", addr["country_code"])]
+            )
+            del addr["country_code"]
+            addr["parent_id"] = partner_id
+            addr["type"] = type_di
+            res_partner_virtual = res_partner_obj.new(addr)
+            # on create res.partner Odoo rewrites address values to be the
+            # same as the parent's, thus we force set to our values
+            for k, v in addr.items():
+                setattr(res_partner_virtual, k, v)
+            version = res_partner_virtual.get_address_version()
+            version.active = True
+            so_vals[field] = version.id
+            del so_vals[vals_key]
+
+    def process_lines(self, so_vals):
+        lines = so_vals["lines"]
+        so_vals["order_line"] = list()
+        for line in lines:
+            product_id = (
+                self.env["product.product"]
+                .search([("default_code", "=", line["product_code"])])
+                .id
+            )
+            qty = line["qty"]
+            price_unit = line["price_unit"]
+            description = line.get("description") or None
+            discount = line["discount"]
+            line_vals_dict = {
+                "product_id": product_id,
+                "product_uom_qty": qty,
+                "price_unit": price_unit,
+                "name": description,
+                "discount": discount,
+            }
+            line_vals_command = (0, 0, line_vals_dict)
+            so_vals["order_line"].append(line_vals_command)
+        del so_vals["lines"]
+
+    def process_amount(self, so_vals):
+        for k, v in so_vals["amount"].items():
+            si_key = "si_" + k
+            so_vals[si_key] = v
+        del so_vals["amount"]
+
+    def process_invoice(self, so_vals):
+        # TODO actually use that val
+        del so_vals["invoice"]
+
+    def process_sale_channel(self, so_vals):
+        channel = self.env["sale.channel"].search(
+            [("name", "=", so_vals.get("sale_channel"))]
+        )
+        if channel:
+            so_vals["sale_channel_id"] = channel.id
+            del so_vals["sale_channel"]
+
+    def simulate_onchanges(self, order):
```

Comment:
can you take a look to this : https://github.com/OCA/server-tools/tree/12.0/onchange_helper
I think it will simply this code